app/renderer/bookmarks.js

```diff
@@ -0,0 +1,512 @@
+ */
+const addFolder = path => {
+    const currentStoredPaths = bookmarkData.folders.map(f => f.path)
+    if (!currentStoredPaths?.includes(path)) {
+        bookmarkData.folders.push(
+            {"keywords": [], "name": "", path})
+    }
+}
+
+
+/**
+ * Write JSON to file.
+ */
+const writeBookmarksToFile = () => {
+    writeJSON(bookmarksFile, bookmarkData)
+}
+
+/**
+ * Fix bookmarkData values based on the option type.
+ * @param {string} option - The option to fix.
+ * @param {string} value - The value to fix.
+ */
+const fixBookmarkData = (option, value) => {
+    let correctFormat = ""
+    if (option === "path") {
+        if (value.startsWith("/")) {
+            correctFormat = value
+        } else {
+            correctFormat = `/${value}`
+        }
+        correctFormat = correctFormat.replace(/\s/g, "-").replace(/[/]$/, "")
+    } else if (option === "keywords") {
+        correctFormat = value.split("").map(k => k.replace(specialChars, "")).
+            join("")
+    } else if (option === "name") {
+        correctFormat = value.replace(specialCharsAllowSpaces, "")
+    } else {
+        correctFormat = value
+    }
+    return correctFormat
+}
+
+/**
+ * Prepare bookmark's object.
+ * @param {string[]} input - The input to parse.
+ * @returns {Bookmark}
+ */
+const bookmarkObject = input => {
+    /** @type {Bookmark} */
+    const newbookmark = {}
+    newbookmark.id = bookmarkData.lastId
+    // If there's "=" or "~" we parse the input.
+    if (input.join().includes("=") || input.join().includes("~")) {
+        // Retain spaces for multiple word inputs.
+        const options = input.join(" ").split("~")
+        for (let i = 0; i < options.length; i++) {
+            // Get key and value: [0,1]
+            const [key, value] = options[i].split("=")
+            if (key === "keywords") {
+                const keywordList = value.split(",").map(
+                    item => item.trim()).filter(Boolean)
+                newbookmark[key] = keywordList
+            } else {
+                const allValue = options[i].split("=").slice(1).join("")
+                if (allValue?.trim()) {
+                    const correctData = fixBookmarkData(key,
+                        allValue)
+                    newbookmark[key] = correctData
+                }
+            }
+        }
+    } else {
+        // If there's only text, we take it as name for current page bookmark
+        newbookmark.name = input.join(" ")
+    }
+    return newbookmark
+}
+
+/**
+ * Check if bookmark is valid.
+ * @param {Bookmark} bookmark - The bookmark object to validate.
+ * @returns {boolean} - True if the bookmark is valid.
+ */
+const isBookmarkValid = bookmark => {
+    let isValid = true
+    const badOptions = []
+    if (!bookmark.name?.trim()) {
+        isValid = false
+        notify({
+            "id": "bookmarks.noname",
+            "src": "user",
+            "type": "dialog"
```

Comment:
These notifications should not use the "dialog" type, instead they should be "warning".